__test__/utils/scraper.test.ts

```diff
@@ -0,0 +1,173 @@
+      };
+
+      fetch.mockImplementation(() =>
+        Promise.resolve({
+          json: () => Promise.resolve(dummySerpApiResponse),
+        })
+      );
+
+      const matchedResult = {
+        title: matchedOrganicResult.title,
+        url: matchedOrganicResult.link,
+        position: matchedOrganicResult.position,
+      };
+
+      const lastResult: SearchResult[] = [matchedResult];
+
+      for (const keyword of getDummyKeywords(lastResult)) {
+        const refreshedResults = await scrapeKeywordFromGoogle(
+          keyword,
+          getDummySettings()
+        );
+
+        const expectedRefreshedResults = {
+          ID: keyword.ID,
+          keyword: keyword.keyword,
+          result: keyword.lastResult,
+
+          position: matchedResult.position,
+          url: matchedResult.url,
+
+          error: false,
+        };
+
+        expect(refreshedResults).toEqual(expectedRefreshedResults);
+      }
+    });
+  });
+
+  describe("When scraper client gets an error response", () => {
+    beforeEach(() => {
+      fetch.mockImplementation(() =>
+        Promise.resolve({
+          json: () => Promise.resolve({ error: "Test error" }),
+        })
+      );
+    });
+
+    it("Returns an error in refreshed results", async () => {
+      for (const keyword of getDummyKeywords()) {
+        const refreshedResults = await scrapeKeywordFromGoogle(
+          keyword,
+          getDummySettings()
+        );
+
+        expect(refreshedResults).toEqual(
+          expect.objectContaining({
+            error: "Test error",
+          })
+        );
+      }
+    });
+  });
+
+  describe("When scraper client gets an unknown response", () => {
+    beforeEach(() => {
+      fetch.mockImplementation(() =>
+        Promise.resolve({
+          json: () => Promise.resolve({ datum: [{ title: "Test title" }] }),
+        })
+      );
+    });
+
+    it("Returns an error in refreshed results", async () => {
+      for (const keyword of getDummyKeywords()) {
+        const refreshedResults = await scrapeKeywordFromGoogle(
+          keyword,
+          getDummySettings()
+        );
+
+        expect(refreshedResults).toMatchObject(
+          expect.objectContaining({
+            error: UNKNOWN_ERROR,
+          })
+        );
+      }
+    });
+  });
+
+  describe("When scraper client gets an unknown SerpApi response", () => {
+    beforeEach(() => {
+      fetch.mockImplementation(() =>
+        Promise.resolve({
+          json: () => Promise.resolve({ search_metadata: { id: 42 } }),
+        })
+      );
+    });
+
+    it("Returns an error in refreshed results", async () => {
+      for (const keyword of getDummyKeywords()) {
+        const refreshedResults = await scrapeKeywordFromGoogle(
+          keyword,
+          getDummySettings()
+        );
+
+        expect(refreshedResults).toEqual(
+          expect.objectContaining({
+            error: UNKNOWN_ERROR,
+          })
+        );
+      }
+    });
+  });
+});
```

Comment:
This file only tests the SerpAPI Intregation correct? Shouldn't it better if its named `scraper.serpapi.test.ts` instead of `scraper.test.ts` ?

Reply:
Sure, thank you!